python-checks/src/main/java/org/sonar/python/checks/BuiltinShadowingAssignmentCheck.java

```diff
@@ -0,0 +1,192 @@
+    context.registerSyntaxNodeConsumer(Tree.Kind.ASSIGNMENT_EXPRESSION, this::checkAssignmentExpression);
+    context.registerSyntaxNodeConsumer(Tree.Kind.CLASSDEF, this::checkClassDefinition);
+    context.registerSyntaxNodeConsumer(Tree.Kind.FUNCDEF, this::checkFunctionDeclaration);
+    context.registerSyntaxNodeConsumer(Tree.Kind.IMPORT_FROM, ctx ->
+      ((ImportFrom) ctx.syntaxNode()).importedNames().forEach(m -> checkImportedNameAlias(ctx, m)));
+    context.registerSyntaxNodeConsumer(Tree.Kind.PARAMETER_LIST, this::checkParameterList);
+  }
+
+  private void checkAssignmentExpression(SubscriptionContext ctx) {
+    AssignmentExpression assignmentExpression = (AssignmentExpression) ctx.syntaxNode();
+    Name lhsName = assignmentExpression.lhsName();
+    if (shouldReportIssue(lhsName)) {
+      ctx.addIssue(lhsName, String.format(MESSAGE, VRBL_ISSUE_TYPE));
+    }
+  }
+
+  private void checkAssignment(SubscriptionContext ctx) {
+    AssignmentStatement assignment = (AssignmentStatement) ctx.syntaxNode();
+    Tree ancestor = TreeUtils.firstAncestorOfKind(assignment, Tree.Kind.FUNCDEF, Tree.Kind.CLASSDEF);
+    if (ancestor == null || ancestor.is(Tree.Kind.FUNCDEF)) {
+      for (int i = 0; i < assignment.lhsExpressions().size(); i++) {
+        for (Expression expression : assignment.lhsExpressions().get(i).expressions()) {
+          if (shouldReportIssue(expression)) {
+            ctx.addIssue(expression, String.format(MESSAGE, VRBL_ISSUE_TYPE));
+          }
+        }
+      }
+    }
+  }
+
+  private void checkAnnotatedAssignment(SubscriptionContext ctx) {
+    AnnotatedAssignment assignment = (AnnotatedAssignment) ctx.syntaxNode();
+    Tree ancestor = TreeUtils.firstAncestorOfKind(assignment, Tree.Kind.FUNCDEF, Tree.Kind.CLASSDEF);
+    if (ancestor == null || ancestor.is(Tree.Kind.FUNCDEF)) {
+      Expression variable = assignment.variable();
+      Token equalToken = assignment.equalToken();
+      if (equalToken != null && shouldReportIssue(variable)) {
+        ctx.addIssue(variable, String.format(MESSAGE, VRBL_ISSUE_TYPE));
+      }
+    }
+  }
+
+  private void checkFunctionDeclaration(SubscriptionContext ctx) {
+    FunctionDef functionDef = (FunctionDef) ctx.syntaxNode();
+    if (!functionDef.isMethodDefinition() && shouldReportIssue(functionDef.name())) {
+      ctx.addIssue(functionDef.name(), String.format(MESSAGE, "function"));
+    }
+  }
+
+  private void checkParameterList(SubscriptionContext ctx) {
+    ParameterList parameterList = (ParameterList) ctx.syntaxNode();
+    if (reportOnParameters && !checkIfMethodIsOverride(parameterList)) {
+      for (Parameter parameter : parameterList.nonTuple()) {
+        Name parameterName = parameter.name();
+        if (parameterName != null && isBuiltInName(parameterName)) {
+          ctx.addIssue(parameterName, String.format(MESSAGE, "parameter"));
+        }
+      }
+    }
+  }
+
+  private static boolean checkIfMethodIsOverride(ParameterList parameterList) {
```

Comment:
_Minor_: could you rename this into something like `isParentAMethodOverride` ? Instinctively I think a method named `checkSomething` is about raising an issue if some condition is met rather than return a boolean.